protocol/messenger_messages.go

```diff
@@ -518,3 +518,52 @@ func (m *Messenger) SendGroupChatMessage(request *requests.SendGroupChatMessage)
 
 	return m.sendChatMessage(context.Background(), message)
 }
+
+func (m *Messenger) DeleteAllCommunityMemberMessages(member string, communityId string, deletedBy string, clock uint64) (*MessengerResponse, error) {
+	messagesToDelete, err := m.persistence.getCommunityMemberAllNonDeletedMessages(member, communityId, clock)
+	if err != nil {
+		return nil, err
+	}
+
+	for i := 0; i < len(messagesToDelete); i++ {
+		messagesToDelete[i].Deleted = true
+		messagesToDelete[i].DeletedBy = deletedBy
+	}
+
+	err = m.persistence.SaveMessages(messagesToDelete)
+	if err != nil {
+		return nil, err
+	}
+
+	response := &MessengerResponse{}
+	for _, messageToDelete := range messagesToDelete {
+		response.AddMessage(messageToDelete)
+		response.AddRemovedMessage(&RemovedMessage{MessageID: messageToDelete.ID, ChatID: messageToDelete.ChatId, DeletedBy: deletedBy})
```

Comment:
I can see this failing if we ban someone with thousands of messages. It's probably not a good idea to return that big of a response.

Maybe we can handle it by just passing the user ID and the community ID and let the frontend remove the messages?
Or ask the front-end to reset the message list and re-fetch from the backend?
Or I guess, if we want to keep it simple, just doing `AddRemovedMessage` is enough? Is it really useful to pass the whole message in the response?

Reply:
Refactored. We do not send full msg, only msgID and chatID which should be ok, I think